transcom/mymove
# [MB-7474] Edit Service Info page

## Description

This PR adds the Edit Service Info page at the `/moves/review/edit-service-info` route, as well as the `ServiceInfoForm` component. The form is a combination of the service member Name, DOD Info, and Current Duty Station forms, and uses the same validation schema as these. 

## Reviewer Notes

To test:
- log in as a service member with an unsubmitted move and navigate to `/moves/review/edit-service-info`.
- verify you can make changes in this form and save them. Changes should be visible on the home screen.
- If you change "Current Duty Station" to the same as the destination duty station for your orders, you should see an error message
- If you change your Rank value, you should see a flash message saying your entitlement has changed.

Questions:
- Where should this page link from? Where should the buttons go back to?
- Should access be restricted, ie if the move has been submitted already? Is there a different state if move has been submitted?

## Setup

To run the app locally:
```sh
make server_run
make client_run
```

You can also view the Storybook component at **Customer Components / Forms / Service Info Form**:
```
yarn storybook
```

## Code Review Verification Steps

* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7474) for this change

## Screenshots

![image](https://user-images.githubusercontent.com/2723066/114596827-b5222880-9c55-11eb-8263-16d5ec660f49.png)

![image](https://user-images.githubusercontent.com/2723066/114596870-bfdcbd80-9c55-11eb-92c1-733a407bb98a.png)
